deploy/scripts/src/btm/test.ts

```diff
@@ -102,21 +204,67 @@ const awsConfig = new AWSConfig({
 })
 
 const eventData = {
-  payload: __ENV.DATA_BTM_SQS_PAYLOAD
+  payloadEventsString: __ENV.DATA_BTM_SQS_PAYLOAD_EVENTS,
+  payloadTimestamp: __ENV.DATA_BTM_SQS_PAYLOAD_TIMESTAMP
 }
 
+const payloadEventsArray = JSON.parse(eventData.payloadEventsString)
+
+const payloadTimestampArray = eventData.payloadTimestamp.split(',')
+const payloadTimestampMin: number = Number(payloadTimestampArray[0])
+const payloadTimestampMax: number = Number(payloadTimestampArray[1])
+
 const sqs = new SQSClient(awsConfig)
 
-export function sendEvent (): void {
-  const messageBody = eventData.payload.replace('UUID', uuidv4())
-  sqs.sendMessage(env.sqs_queue, messageBody)
+export function sendEventType1 (): void {
+  let messageBody: string = ''
+  messageBody = JSON.stringify(payloadEventsArray[0])
+  sendSQSMessage(messageBody)
+}
+
+export function sendEventType2 (): void {
+  let messageBody: string = ''
+  messageBody = JSON.stringify(payloadEventsArray[1])
+  sendSQSMessage(messageBody)
+}
+
+export function sendEventType3 (): void {
+  let messageBody: string = ''
+  messageBody = JSON.stringify(payloadEventsArray[2])
+  sendSQSMessage(messageBody)
+}
+
+export function sendEventType4 (): void {
+  let messageBody: string = ''
+  messageBody = JSON.stringify(payloadEventsArray[3])
+  sendSQSMessage(messageBody)
+}
+
+export function sendEventType5 (): void {
+  let messageBody: string = ''
+  messageBody = JSON.stringify(payloadEventsArray[4])
+  sendSQSMessage(messageBody)
+}
+
+export function sendEventType6 (): void {
+  let messageBody: string = ''
+  messageBody = JSON.stringify(payloadEventsArray[5])
+  sendSQSMessage(messageBody)
+}
+
+export function sendEventType7 (): void {
+  let messageBody: string = ''
+  messageBody = JSON.stringify(payloadEventsArray[6])
+  sendSQSMessage(messageBody)
 }
 
-export function sendEventDebug (): void {
-  const messageBody = eventData.payload.replace('UUID', uuidv4())
+export function sendSQSMessage (messageBody: string): void {
+  let randomTimestamp: number = 0
+  let timestampFormatted: string = ''
+  randomTimestamp = randomIntBetween(payloadTimestampMin, payloadTimestampMax)
+  timestampFormatted = new Date(randomTimestamp * 1000).toISOString()
+  messageBody = messageBody.replace('TIMESTAMP', randomTimestamp.toString())
+  messageBody = messageBody.replace('TIMESTAMP_FORMATTED', timestampFormatted.replace('Z', ''))
+  messageBody = messageBody.replace('UUID', uuidv4())
   sqs.sendMessage(env.sqs_queue, messageBody)
-  console.log('1 === debug === env.sqs_queue', env.sqs_queue)
-  console.log('2 === debug === awsConfig', awsConfig)
-  console.log('3 === debug === payload', eventData.payload)
-  console.log('4 === debug === messageBody', messageBody)
 }
```

Comment:
I think this'll do it

```
export const [
  sendEventType1,
  sendEventType2,
  sendEventType3,
  sendEventType4,
  sendEventType5,
  sendEventType6,
  sendEventType7
] = payloadEventsArray.map(payload => () => {sendSQSMessage(payload)})

export function sendSQSMessage (messageBody: Record<string, unknown>): void {
  const payloadTimestampArray = eventData.payloadTimestamp.split(',')
  const payloadTimestampMin: number = Number(payloadTimestampArray[0])
  const payloadTimestampMax: number = Number(payloadTimestampArray[1])

  const randomTimestamp = Math.floor(Math.random() * (payloadTimestampMax - payloadTimestampMin + 1)) + payloadTimestampMin
  const timestampFormatted = new Date(randomTimestamp * 1000).toISOString().replace('Z', '')

  messageBody.timestamp = randomTimestamp
  messageBody.timestamp_formatted = timestampFormatted
  messageBody.event_id = uuidv4()

  sqs.sendMessage(env.sqs_queue, JSON.stringify(messageBody))
}
```

Reply:
NB no mutable variables in the global scope (ie `let messages = ''` etc), easier to expand when new payloads are added you can just tack `sendEventType8` onto that deconstructive declaration, passing the messageBody to `sendSQSMessage` as an object and stringifying it later makes it easier and more reliable to to change the properties. Hope that helps

Reply:
Oh, it's worth linting this, I wrote it blind

Reply:
Thanks for the review, have updated the code as suggested to pass to the sendSQSMessage as object. Can look at how to address the load scenarios to functions calls using map on a separate PR as there are some K6 code limitations to address.